modules/swagger-codegen/src/main/resources/go/api.mustache

```diff
@@ -43,6 +43,32 @@ func (a {{classname}}) {{nickname}} ({{#allParams}}{{paramName}} {{{dataType}}}{
 
     _sling := sling.New().{{httpMethod}}(a.Configuration.BasePath)
 
+    {{#authMethods}}// authentication ({{name}}) required
+    {{#isApiKey}}{{#isKeyInHeader}}
+    // set key with prefix in header
+    _sling.Set("{{keyParamName}}", a.Configuration.GetApiKeyWithPrefix("{{keyParamName}}")
+    {{/isKeyInHeader}}{{#isKeyInQuery}}
+    // set key with prefix in querystring
+    {{#hasKeyParamName}}    type KeyQueryParams struct {
```

Comment:
@wing328 code has been updated, but there is no sample that uses query param, so i am not 100% sure if the output will be correct.